rfcs/text/0012-orchestrator-field-set.md

```diff
@@ -19,9 +19,68 @@ add a new schema rather than change existing material.
 
 ## Fields
 
-<!--
-Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
--->
+The proposed change adds six fields, as described below:
+
+```
+---
+- name: orchestrator
+  title: Orchestrator
+  group: 2
+  short: Fields relevant to container orchestrators.
+  description: >
+    Fields that describe the resources which container orchestrators manage or
+    act upon.
+  type: group
+  fields:
+    - name: cluster
+      level: core
+      type: keyword
+      example: production
+      description: >
+        Orchestrator cluster name.
+
+    - name: type
+      level: core
+      type: keyword
+      example: Nomad, Kubernetes
+      description: >
+        Orchestrator cluster type.
+
+    - name: namespace
+      level: core
+      type: keyword
+      example: kube-system
+      description: >
+        Namespace in which the action is taking place.
+
+    - name: resource.name
+      level: core
+      type: keyword
+      example: test-pod-cdcws
+      description: >
+        Name of the resource being acted upon.
+
+    - name: resource.type
+      level: core
+      type: keyword
+      example: service
+      description: >
+        Type of resource being acted upon.
+
+    - name: subresource.type
+      level: extended
+      example: status
+      type: keyword
+      description: >
+        Subresource of the resource being acted upon, if applicable.
```

Comment:
What would be the use of this field? This may be too dependant on the implementation. Even in the same orchestrator it may be dependant on the kind of resource (e.g. a Deployment might have as subresources the replicaset, the pod, their statuses...).

Reply:
Kubernetes subresources are a poorly defined concept - I can't find any good documentation to point to, but see [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#subresources) and [here](https://stackoverflow.com/questions/49396607/where-can-i-get-a-list-of-kubernetes-api-resources-and-subresources) for some examples. Basically, it's perhaps something that can be done to a resource - technically a `pod` is not a subresource of a `deployment`.

For pods, the specific list of subresources is: `attach`, `binding`, `eviction`, `exec`, `log`, `portforward`, `proxy`, and `status` - at least on a vanilla microk8s installation. You can see the full list for your flavour of k8s by following the instructions in the second link above.

There is an alternative representation of subresources: they can be represented alongside the resources themselves, as `pods/attach` for example. So it's perhaps not essentials that they exist as a separate entity in this fieldset.

I am not sure if the resource/subresource distinction translates at all to other orchestrators, so I'm happy to remove this field if it is too specific.

Reply:
> There is an alternative representation of subresources: they can be represented alongside the resources themselves, as `pods/attach` for example. So it's perhaps not essentials that they exist as a separate entity in this fieldset.

Perhaps to identify events related to subresources we should use specific fields, for example for a pod's container metrics, the `container` fields could be used:
```
orchestrator.type: kubernetes
orchestrator.resource: pod
container.name: somecontainer
```
Same thing might happen with other relevant subresources.

So I think we can go on without `orchestrator.subresource` by now and try to use other fields for relevant subresources. In the future we can still add it if we find a better definition.